### PR TITLE
Rearrange code, optimized something

### PR DESCRIPTION
**Pull request by K4zoku**
- Using memmove, memcpy instead of linear copy
- add prefix __ for internal function
- Beautify code: 
  - Using space instead of tab
  - The backslashes are now all in the column 120

### CODE REVIEW

**Comment by andy5995** [comment]
@K4zoku Does this work with the new tests I added recently? ('make check' and 'make memcheck')

**Comment by andy5995** [comment]
> @K4zoku Does this work with the new tests I added recently? ('make check' and 'make memcheck')

Actually, yes it does. :) I just tried it...

Also with -Wall and -Wextra flags, no warnings.

```
[andy@prometheus build]$ make VERBOSE=1
/usr/bin/cmake -S/home/andy/src/c-vector -B/home/andy/src/c-vector/build --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/cmake -E cmake_progress_start /home/andy/src/c-vector/build/CMakeFiles /home/andy/src/c-vector/build//CMakeFiles/progress.marks
make  -f CMakeFiles/Makefile2 all
make[1]: Entering directory '/home/andy/src/c-vector/build'
make  -f CMakeFiles/c-vector-example.dir/build.make CMakeFiles/c-vector-example.dir/depend
make[2]: Entering directory '/home/andy/src/c-vector/build'
cd /home/andy/src/c-vector/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/andy/src/c-vector /home/andy/src/c-vector /home/andy/src/c-vector/build /home/andy/src/c-vector/build /home/andy/src/c-vector/build/CMakeFiles/c-vector-example.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/home/andy/src/c-vector/build'
make  -f CMakeFiles/c-vector-example.dir/build.make CMakeFiles/c-vector-example.dir/build
make[2]: Entering directory '/home/andy/src/c-vector/build'
[ 50%] Building C object CMakeFiles/c-vector-example.dir/example.c.o
/usr/bin/cc   -Wall -Wextra -MD -MT CMakeFiles/c-vector-example.dir/example.c.o -MF CMakeFiles/c-vector-example.dir/example.c.o.d -o CMakeFiles/c-vector-example.dir/example.c.o -c /home/andy/src/c-vector/example.c
[100%] Linking C executable c-vector-example
/usr/bin/cmake -E cmake_link_script CMakeFiles/c-vector-example.dir/link.txt --verbose=1
/usr/bin/cc -rdynamic CMakeFiles/c-vector-example.dir/example.c.o -o c-vector-example 
make[2]: Leaving directory '/home/andy/src/c-vector/build'
[100%] Built target c-vector-example
make[1]: Leaving directory '/home/andy/src/c-vector/build'
/usr/bin/cmake -E cmake_progress_start /home/andy/src/c-vector/build/CMakeFiles 0
```

**Comment by eteran** [comment]
@K4zoku

Some comments:

1. I prefer the style of ` if (src) {` over ` if (src != NULL) {`
2. a leading double underscore (`__`) is technically non-compliant C code|
3. I'm not sure that I want to change the style from tabs to spaces. I understand that many prefer spaces, but it's not really for me. I prefer tabs for indentation, but spaces for alignment. In otherwords, in my style, "a tab will be preceeded by a tab or nothing"

I certainly appreciate the optimizations, but I'd like to keep them seperate from any "style" based changes since they are subject to (often string) opinions.

**Comment by eteran** [comment]
Also changing the order from `from`/`to` to `dest`/`src` is a change I'd have to think about since it would break any existing users shoudl they try to update to the latest version of this.

**Comment by eteran** [comment]
Also, looks like you put two implementations of `cvector_insert` in there ;-)

**Comment by eteran** [comment]
@K4zoku This is looking great now, thanks for the changes! I'll take a closer look in the morning since it's getting pretty late for me here. I may end up adding a clang-format file to the repo since you're not the first person to suggest stylistic changes.

That being said, I am going to give the space vs tabs considering given that this is a "public file" and we want it to look good in all editors, both online and offline... not just my editor with my settings, so ... we'll see. how I feel about the spaces, i may come around ;-)

**Comment by K4zoku** [comment]
ok, looks like everything is fine now

**Comment by eteran** [comment]
Great, will be looking this PR soon  👍🏻

File: test.c [hunk 1]
```diff
@@ -63,5 +64,59 @@ int main() {
 	/* well, we don't have destructors, so let's clean things up */
 	cvector_free(v);
 
+    putchar('\n');
```
Comment (andy5995) [inline test.c:67]: Just curious... is that generally a little better to use than `puts ("")`

**Pull request merged**